Modules/Devices/src/Devices.Domain/Entities/Device.cs

```diff
@@ -46,20 +46,23 @@ public Device(Identity identity, DeviceId? createdByDevice = null)
     public static Expression<Func<Device, bool>> IsNotDeleted =>
         device => device.DeletedAt == null && device.DeletedByDevice == null;
 
-    public DomainError? CanBeDeleted(DeviceId deletedByDevice)
+    public DomainError? CanBeDeleted(DeviceId deletedByDevice, IdentityAddress identityAddress)
     {
         if (IsOnboarded)
             return new DomainError("error.platform.validation.device.deviceCannotBeDeleted", "The device cannot be deleted because it is already onboarded.");
 
-        if (Id == deletedByDevice)
-            return new DomainError("error.platform.validation.device.deviceCannotBeDeleted", "The device cannot be deleted.");
+        if (deletedByDevice == Id)
+            return new DomainError("error.platform.validation.device.deviceCannotBeDeleted", "The device cannot be deleted by itself.");
```

Comment:
We agreed on deleting this check because it is not necessary, as in such a case the first check would already fail.